src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -671,15 +672,15 @@ private static XmlSerializer[] GetSerializersFromCache(XmlMapping[] mappings, Ty
                     foreach (XmlSerializerMappingKey mappingKey in pendingKeys.Keys)
                     {
                         index = pendingKeys[mappingKey];
-                        serializers[index] = (XmlSerializer)contract.TypedSerializers[mappingKey.Mapping.Key];
-                        serializers[index].SetTempAssembly(tempAssembly, mappingKey.Mapping);
+                        serializers[index] = (XmlSerializer)contract.TypedSerializers[mappingKey.Mapping.Key!]!;
```

Comment:
Same as before.

Reply:
it will throw NRE on next line if it is null